### ISSUE TITLE
C++ Expert wanted ...

### ISSUE THREAD

**Issue by mitra42** [mitra42/frugal-iot#94]
There are a few challenges that an expert C++ coder could help with .

- [ ] In actuator_digital.cpp - and actuator_ledbuiltin.cpp - I'd like to pass a callback that is a method of an instance of a class, but I'm completely failing to get something that will compile. 
- [ ]

**Comment by jksemple** [comment]
If you subclass all your different actuators from a common super Actuator which has a virtual inputReceived() method you can then pass across a reference to the actuator object instead of passing the callback.  That can then be held in your subscriptions collection and then the MQTT object can call actuator.inputReceived() on each such subscribed actuator object.  The appropriate concrete inputReceived() method will then get executed. I think that might work.

**Comment by mitra42** [comment]
Yes - it might, the challenge is that about half of the sensors/actuators/control are not done as Classes - they are using the "setup/loop" paradigm of Arduino IDE.   Its a gradual rewrite to turn them into classes.

**Comment by jksemple** [comment]
If you don't want to rewrite all your actuators as classes you'll just have to stick to the callback approach and where you have a class you'll need to define a callback function outside the class that calls this->inputReceived(). Not pretty, but you can't pass an object method as a function pointer parameter AFAIK.

**Comment by mitra42** [comment]
Ok - its quite possibly the time to bite the bullet and rewrite as classes, not just for this reason.  I've got a issue for this somewhere and will stick a proposed class hierarchy there after the break. #25

**Comment by jksemple** [comment]
In sensor.cpp you wanted to declare sensors as a static member of Sensor.  You can do this but you do need to initialise the static member before 'main' runs. Somewhere in sensor.cpp add the line std::vector<Sensor*> Sensor::sensors = std::vector<Sensor*>();
